lib/wise_homex/json_parser.ex

```diff
@@ -256,13 +259,22 @@ defmodule WiseHomex.JSONParser do
   end
 
   defp set_struct_attribute(struct, key, value) do
-    if Map.has_key?(struct, key) do
-      %{struct | key => value}
-    else
-      struct
+    %{__struct__: module} = struct
+    fields = module.__changeset__()
+
+    case Map.fetch(fields, key) do
+      {:ok, type} when type in [:date, :utc_datetime] -> coerce_struct_value(struct, key, value)
+      {:ok, _} -> %{struct | key => value}
+      _ -> struct
     end
   end
 
+  defp coerce_struct_value(struct, key, value) do
```

Comment:
I think this function is difficult to read. I think it should have another function name, and/or good comments describing its purpose.

Definition:
_Coerce: To pressure, intimidate, or force (someone) into doing something_

Reply:
I think "Type coercion" is the normal term used to force a value to a different type. E.g. a string containing a date to an actual Date. It is even mentioned on Wikipedia as an alias of "Type conversion" :)

Agree to add comment about what it is doing.

Reply:
Good to know. I didn't know that term.

Reply:
I know it best from Ruby-land. Did a quick seach in ecto and only found some issues regarding "coercion" or "coerce".

Reply:
Fixed in 77025602baaf9a0366655aba754770fe9da58ace